docs/internal/DistributedArchitectureGuide.md

```diff
@@ -188,15 +188,80 @@ Some concepts are applicable to both cluster and project scopes, e.g. [persisten
 
 ### Translog
 
-(Explain checkpointing and generations, when happens on Lucene flush / fsync)
+[Basic write model]:https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-replication.html
+[Translog]:https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/translog/Translog.java
+[History retention]:https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-history-retention.html
 
-(Concurrency control for flushing)
+It is important to understand first the [Basic write model] of documents:
+documents are written to Lucene in-memory buffers, then "refreshed" to searchable segments which may not be persisted on disk, and finally "flushed" to a durable Lucene commit on disk.
+This means that newly ingested data may be lost if there is an outage before the next persisted commit on disk.
+For this reason, newly ingested data is also written to a shard's [Translog], whose main purpose is to persist uncommitted operations (e.g., document insertions or deletions), so they can be replayed in the event of ephemeral failures such as a crash or power loss.
+The translog is always persisted and fsync'ed on disk before acknowledging writes back to the user.
+This can be seen in `InternalEngine` which calls the `add()` method of the translog to append operations, e.g., its `index()` method at some point adds a document insertion operation to the translog.
+The translog ultimately truncates operations once they have been flushed to disk by a Lucene commit.
 
-(VersionMap)
+Main usages of the translog are:
+
+* During recovery, an index shard can be recovered up to at least the last acknowledged operation by replaying the translog onto the last flushed commit of the shard.
+* During a replica recovery, it may recover some lost operations from the primary's translog if needed before falling back to a complete recovery of Lucene files from the primary.
```

Comment:
Is this true, I thought I read something to the contrary? https://elastic.slack.com/archives/C06FWH2BGMS/p1741793855287379?thread_ts=1741787680.838199&cid=C06FWH2BGMS